src/instantsend/instantsend.cpp

```diff
@@ -379,22 +389,35 @@ MessageProcessingResult CInstantSendManager::ProcessInstantSendLock(NodeId from,
     const CBlockIndex* pindexMined{nullptr};
     const bool found_transaction{tx != nullptr};
     // we ignore failure here as we must be able to propagate the lock even if we don't have the TX locally
+    int minedHeight{-1};
     if (found_transaction && !hashBlock.IsNull()) {
-        pindexMined = WITH_LOCK(::cs_main, return m_chainstate.m_blockman.LookupBlockIndex(hashBlock));
-
-        // Let's see if the TX that was locked by this islock is already mined in a ChainLocked block. If yes,
-        // we can simply ignore the islock, as the ChainLock implies locking of all TXs in that chain
-        if (pindexMined != nullptr && clhandler.HasChainLock(pindexMined->nHeight, pindexMined->GetBlockHash())) {
-            LogPrint(BCLog::INSTANTSEND, "CInstantSendManager::%s -- txlock=%s, islock=%s: dropping islock as it already got a ChainLock in block %s, peer=%d\n", __func__,
-                     islock->txid.ToString(), hash.ToString(), hashBlock.ToString(), from);
-            return {};
+        if (auto h = GetBlockHeight(hashBlock)) {
+            minedHeight = *h;
+            // Let's see if the TX that was locked by this islock is already mined in a ChainLocked block. If yes,
+            // we can simply ignore the islock, as the ChainLock implies locking of all TXs in that chain
+            if (clhandler.HasChainLock(minedHeight, hashBlock)) {
+                LogPrint(BCLog::INSTANTSEND, "CInstantSendManager::%s -- txlock=%s, islock=%s: dropping islock as it already got a ChainLock in block %s, peer=%d\n", __func__,
+                         islock->txid.ToString(), hash.ToString(), hashBlock.ToString(), from);
+                return {};
+            }
```

Comment:
This block is repeated twice. We should calculate `minedHeight` via all possible ways first and only then check if it's chainlocked.